sklearn/tree/_criterion.pyx

```diff
@@ -268,6 +312,12 @@ cdef class ClassificationCriterion(Criterion):
                 self.sum_right == NULL):
             raise MemoryError()
 
+        if allow_missing:
+            self.sum_missing = <double*> calloc(n_elements, sizeof(double))
+            self.sum_available = <double*> calloc(n_elements, sizeof(double))
+            if self.sum_missing == NULL or self.sum_available == NULL:
+                raise MemoryError()
```

Comment:
Why `safe_realloc`? The `sum_missing` is not initialized yet (like `sum_right` above it is initialized only once...)

Maybe I'm missing something...